src/index.js

```diff
@@ -9,20 +9,24 @@ export default function concatMap(project) {
     const innerSink = (type, data) => {
       if (type === 0) {
         innerTalkback = data
-        innerTalkback(1)
         return
       }
 
       if (type === 1) {
         sink(1, data)
-        innerTalkback(1)
+        return
+      }
+
+      if (type === 2 && data) {
+        sink(2, data)
         return
       }
 
       if (type === 2) {
         innerTalkback = null
 
         if (queue.length === 0) {
+          sink(2)
```

Comment:
this doesnt seem right - we should only complete the sink if the queue is empty *and* the outer source is completed